docs/documents.md

```diff
@@ -81,6 +82,54 @@ doc.name = 'foo';
 await doc.save(); // Throws DocumentNotFoundError
 ```
 
+## Setting Nested Properties
+
+Mongoose documents have a `set()` function that you can use to safely set deeply nested properties.
+
+```javascript
+const schema = new Schema({
+  nested: {
+    subdoc: new Schema({
+      name: String
+    })
+  }
+});
+const TestModel = mongoose.model('Test', schema);
+
+const doc = new TestModel();
+doc.set('nested.subdoc.name', 'John Smith');
+doc.nested.subdoc.name; // 'John Smith'
+```
+
+Mongoose documents also have a `get()` function that lets you safely read deeply nested properties. `get()` lets you avoid having to explicitly check for nullish values, similar to JavaScript's [optional chaining operator `?.`](https://masteringjs.io/tutorials/fundamentals/optional-chaining-array).
+
+```javascript
+const doc2 = new TestModel();
+
+doc2.get('nested.subdoc.name'); // undefined
+doc2.nested?.subdoc?.name; // undefined
+
+doc2.set('nested.subdoc.name', 'Will Smith');
+doc2.get('nested.subdoc.name'); // 'Will Smith'
+```
+
+You can use optional chaining `?.` and nullish coalescing `??` with Mongoose documents.
+However, be careful when using [nullish coalescing assignments `??=`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) to create nested paths with Mongoose documents.
+
+```javascript
+// The following works fine
+const doc3 = new TestModel();
+doc3.nested.subdoc ??= {};
+doc3.nested.subdoc.name = 'John Smythe';
+
+// The following does **NOT** work.
+// Do not use the following pattern with Mongoose documents.
+const doc4 = new TestModel();
+(doc4.nested.subdoc ??= {}).name = 'Charlie Smith';
```

Comment:
does this not work because mongoose is cloning on assignment? (i have tried in a node REPL)

```js
ret = (doc4.nested.subdoc ??= {})
ret === doc4.nested.subdoc // eval "false"
```

Reply:
Yes. The problem is the following:

```javascript
const x = {};
(doc4.nested.subdoc ??= x) === x; // true
doc4.nested.subdoc === x; // false
```

Basically `a.b ??= x` evaluates to `x`, not the value of `a.b` after the assignment. Which isn't a big deal in most cases, but with Mongoose the distinction matters.

We need to clone because `doc4.nested.subdoc` needs change tracking. This may be a case where using proxies instead of Object.defineProperty() for change tracking would help.